src/services/carSaleTracking/buyerFeedback.ts

```diff
@@ -0,0 +1,20 @@
+import { postData, ApiCallOptions } from "../../base"
+import { WithValidationError } from "../../types/withValidationError"
+
+export const addPurchaseConfirmation = async ({
+  key,
+  confirmed,
+  options = {},
+}: {
+  key: string
+  confirmed: boolean
+  options?: ApiCallOptions
+}): Promise<WithValidationError> => {
+  return postData({
```

Comment:
IIRC the post and put functions do not abstract the validation handling so you have to wrap the error handling in some try catches to fulfill the interface...
```
 try {
    const result = await putData({
      service: Service.SEARCH,
      path: `dealers/${dealerId}/listing-saved-searches/auto-alarms/${savedSearchId}`,
      body: savedSearch,
      options: { isAuthorizedRequest: true, ...options },
    })

    return {
      tag: "success",
      result,
    }
  } catch (error) {
    return handleValidationError(error, { swallowErrors: true })
  }
```
(Very happy to discuss the envelope approach in general but I think we should not mix patterns for now)

Reply:
I have been thinking about this as well, but as there is no returned data on this call (it's a fire and forget for us) and I have to rely on the response codes to react correctly in the application. As there is no validation required, I went with not wrapping them

Does that make sense? Otherwise I'm happy to discuss in a call! I was back and forth here too

Reply:
we decided to roll with it